src/silo/query_engine/filter_expression/negation.cpp

```diff
@@ -31,7 +31,7 @@ std::unique_ptr<operators::Operator> Negation::compile(
       return child_operator;
    }
    if (child_operator->type() == operators::SELECTION) {
-      dynamic_cast<operators::Selection*>(child_operator.get())->negate();
+      dynamic_cast<operators::Selection<std::string>*>(child_operator.get())->negate();
```

Comment:
I would think that this also breaks if the selection is not of type `std::string`